crates/rpc-types/src/genesis.rs

```diff
@@ -27,3 +79,172 @@ pub struct OptimismBaseFeeInfo {
     /// EIP-1559 denominator after canyon
     pub eip1559_denominator_canyon: Option<u64>,
 }
+
+impl OptimismBaseFeeInfo {
+    /// Extract the Optimism base fee info from `OtherFields` object.
+    pub fn extract_from(others: &OtherFields) -> Option<Self> {
+        if let Some(Ok(optimism_base_fee_info)) =
+            others.get_deserialized::<OptimismBaseFeeInfo>("optimism")
+        {
+            return Some(optimism_base_fee_info);
+        }
+        None
+    }
+}
+
+impl TryFrom<OtherFields> for OptimismBaseFeeInfo {
+    type Error = serde_json::Error;
+
+    fn try_from(others: OtherFields) -> Result<Self, Self::Error> {
+        if let Some(Ok(optimism_base_fee_info)) =
+            others.get_deserialized::<OptimismBaseFeeInfo>("optimism")
+        {
+            Ok(optimism_base_fee_info)
```

Comment:
this code is duplicated, this should reuse the extract fn

Reply:
But if we reuse the extract fn we lose the error information (since extract returns option) - thus cannot use `type Error = serde_json::Error`. or should we define custom error enum?

Reply:
I see, then you can flip it the other way around